core/contracts/Voting.sol

```diff
@@ -27,61 +27,50 @@ contract Voting is Testable, MultiRole, OracleInterface {
     using ResultComputation for ResultComputation.Data;
 
     // Identifies a unique price request for which the Oracle will always return the same value.
+    // Tracks ongoing votes as well as the result of the vote.
     struct PriceRequest {
         bytes32 identifier;
         uint time;
-    }
-
-    struct Round {
-        // The list of price requests that were/are being voted on this round.
-        PriceRequest[] priceRequests;
-
-        // Voting token snapshot ID for this round. If this is 0, no snapshot has been taken.
-        uint snapshotId;
 
-        // Inflation rate set for this round.
-        FixedPoint.Unsigned inflationRate;
-    }
-
-    // A particular voter's submission.
-    struct VoteSubmission {
-        // A bytes32 of `0` indicates no commit or a commit that was already revealed.
-        bytes32 committedHash;
-
-        // The value of the vote that was revealed.
-        int revealedVote;
+        mapping(uint => VoteInstance) votes;
 
-        // Whether the voter revealed their vote (this is to handle valid reveals for 0 prices).
-        bool didReveal;
+        // The final result of a vote.
+        // If lastVotingRound < voteTiming.getLastUpdatedRoundId, vote was resolved.
+        int resolvedPrice;
+        uint lastVotingRound;
     }
 
     struct VoteInstance {
-        // Maps (voterAddress) to their submission.
-        mapping(address => VoteSubmission) voteSubmissions;
+        mapping(address => bytes32) commits;
```

Comment:
I'm indifferent here. I left these as three maps because I want to keep the data structures flat, and I want to push for eventually merging these three maps into one :)

I do get that it's less intuitive though.

Reply:
Actually, I take this back. I'm going to update the data structs.

Reply:
You want to eventually push for merging all three into one? Like into a struct? I'm a little confused...?

I'm assuming you eventually want them all to fit into a single word, but if we were to do that, I think they would still be encapped in a struct rather than an opaque `uint256` to reduce the amount of bitwise math we'd need to manually implement. I think it would look something like:
```
enum VoteSubmissionState {
  None,
  Commit,
  Reveal
}

struct VoteSubmission {
  VoteSubmissionState state;
  int248 commitOrPrice;
}
```

Reply:
Yeah, either an enum or a single word with bitwise operations. It depends on how much the efficiency gains are, of course.